bin/check-dns.rb

```diff
@@ -145,52 +223,17 @@ def check_ips(entries)
 
   def run
     unknown 'No domain specified' if config[:domain].nil?
+    unknown 'Count must be 1 or more' if config[:count] < 1
 
-    begin
-      entries = resolve_domain
-    rescue Dnsruby::NXDomain
-      output = "Could not resolve #{config[:domain]} #{config[:type]} record"
-      critical(output)
-      return
-    rescue => e
-      output = "Could not resolve  #{config[:domain]}: #{e}"
-      config[:warn_only] ? warning(output) : critical(output)
-      return
-    end
-    puts entries.answer if config[:debug]
-    if entries.answer.length.zero?
-      output = "Could not resolve #{config[:domain]} #{config[:type]} record"
-      config[:warn_only] ? warning(output) : critical(output)
-    elsif config[:result]
-      # special logic for checking ipaddresses with result
-      # mostly for ipv6 but decided to use the same logic for
-      # consistency reasons
-      if config[:type] == 'A' || config[:type] == 'AAAA'
-        check_ips(entries)
-      # non ip type
-      else
-        b = if entries.answer.count > 1
-              entries.answer.rrsets(config[:type].to_s).to_s
-            else
-              entries.answer.first.to_s
-            end
-        if b.include?(config[:result])
-          ok "Resolved #{entries.security_level} #{config[:domain]} #{config[:type]} included #{config[:result]}"
-        else
-          critical "Resolved #{config[:domain]} #{config[:type]} did not include #{config[:result]}"
-        end
-      end
-
-    elsif config[:regex]
-      check_against_regex(entries, Regexp.new(config[:regex]))
+    entries = resolve_domain
+    errors, success = check_results(entries)
 
-    elsif config[:validate]
-      if entries.security_level != 'SECURE'
-        critical "Resolved  #{entries.security_level} #{config[:domain]} #{config[:type]}"
-      end
-      ok "Resolved #{entries.security_level} #{config[:domain]} #{config[:type]}"
+    percent = success.count.to_f / config[:count].to_f * 100
```

Comment:
Is there value to casing `config[:count]` as a float? last I checked (float / int * int) will still give you a float.